src/test/java/jimmy/mcgymmy/logic/commands/UndoCommandTest.java

```diff
@@ -0,0 +1,45 @@
+package jimmy.mcgymmy.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+import jimmy.mcgymmy.model.ModelManager;
+
+/**
+ * Contains unit tests for UndoCommand.
+ */
+public class UndoCommandTest {
+    private class CanUndoModelStub extends ModelManager {
+        @Override
+        public boolean canUndo() {
+            return true;
+        }
+
+        @Override
+        public void undo() {
+            // do nothing (to prevent EmptyStackException)
```

Comment:
maybe something more useful here could be to set a boolean flag `hasBeenUndoed` or something to `true` here, and checking it in the `execute_canUndo_success` part

Reply:
Done!